README.md

```diff
@@ -289,6 +290,34 @@ secretDescriptor:
       property: password
 ```
 
+### Hashicorp Vault
+
+kubernetes-external-secrets supports fetching secrets from [Hashicorp Vault](https://www.vaultproject.io/), using the [Kubernetes authentication method](https://www.vaultproject.io/docs/auth/kubernetes.html).
+
+You will need to set the `VAULT_ADDR` environment variables so that kubernetes-external-secrets knows which endpoint to connect to, then create `ExternalSecret` definitions as follows:
+
+```yml
+apiVersion: 'kubernetes-client.io/v1'
+kind: ExternalSecret
+metadata:
+  name: hello-vault-service
+secretDescriptor:
+  backendType: vault
+  # Your authentication mount point, e.g. "kubernetes"
+  vaultMountPoint: my-kubernetes-vault-mount-point
+  # The vault role that will be used to fetch the secrets
+  # This role will need to be bound to kubernetes-external-secret's ServiceAccount; see Vault's documentation:
+  # https://www.vaultproject.io/docs/auth/kubernetes.html
+  vaultRole: my-vault-role
+  data:
+  # The full path of the secret to read, as in `vault read secret/data/hello-service/credentials`
+  - key: secret/data/hello-service/credentials
+    property: password
+  # Properties are matched individually. If you have several properties in your Vault secret, you will need to add them all separately
+  - key: secret/data/hello-service/credentials
+    property: api-key
```

Comment:
I think name always have to be set as well

Reply:
Oh, gotcha - I didn't get this from the docs; will update the code & example 👍